ui/src/routes/auth.test.ts

```diff
@@ -0,0 +1,216 @@
+// Copyright 2024 Defense Unicorns
+// SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+
+import { authenticated } from '$features/auth/store'
+import { beforeEach, describe, expect, vi } from 'vitest'
+
+import { load } from './+layout'
+
+// Mock stores
+vi.mock('$features/k8s/namespaces/store', () => ({
+  createStore: vi.fn(() => ({
+    start: vi.fn(),
+  })),
+}))
+vi.mock('$features/auth/store', () => ({
+  authenticated: {
+    set: vi.fn(),
+    subscribe: vi.fn(() => () => {}),
+    update: vi.fn(),
+  },
+}))
+
+describe('load function', () => {
+  // Mock fetch
+  const fetchMock = vi.fn()
+  global.fetch = fetchMock
+
+  let mockUrl: URL
+
+  beforeEach(() => {
+    vi.clearAllMocks()
+
+    // Reset URL mock
+    mockUrl = new URL('https://example.com')
+    Object.defineProperty(window, 'location', {
+      value: { href: mockUrl.href },
+      writable: true,
+    })
+  })
+
+  test('successful local authentication with token in URL', async () => {
+    // Set up URL with token
+    mockUrl.searchParams.set('token', 'valid-token')
+    Object.defineProperty(window, 'location', {
+      value: { href: mockUrl.href },
+      writable: true,
+    })
+
+    const mockUserData = {
+      name: '',
+      'preferred-username': '',
+      group: '',
+      'in-cluster-auth': false,
+    }
+
+    // Mock successful fetch response
+    fetchMock.mockResolvedValueOnce({
+      ok: true,
+      json: () => Promise.resolve(mockUserData),
```

Comment:
nit more FYI than request to change: looks like you don't need Promise.resolve() for this.
tests still pass as
```suggestion
      json: () => mockUserData,
```

Reply:
done! removed here and in other tests, good catch